### ISSUE TITLE
Wrong distance btw triangle and point

### ISSUE THREAD

**Issue by ATeryohin** [facebookresearch/pytorch3d#698]
I have triangle with vertecies: 
[10.67399979, 8.61225986, 0.77146602]
[10.67269993, 8.61141968, 0.77197701]
[10.67399979, 8.61225986, 0.77197701]
And point with coords:
[5, 5, 5]
I created from this data pytorch3d.structures.meshes.Meshes and pytorch3d.structures.pointclouds.Pointclouds.
Using loss.loss.point_mesh_face_distance i get distance = 0.0042.
Why distance so small?
Or am I using the method incorrectly?

Below code to repeat this behavior:
------------------------------------------------
from pytorch3d import loss
import torch
from pytorch3d.structures.meshes import Meshes as torch_mesh
from pytorch3d.structures.pointclouds import Pointclouds as torch_pc

vertecies = [torch.cuda.FloatTensor([[10.67399979, 8.61225986, 0.77146602], 
[10.67269993, 8.61141968, 0.77197701], 
[10.67399979, 8.61225986, 0.77197701]])]
faces = [torch.cuda.LongTensor([[0, 1, 2]])]
points_tensor = [torch.cuda.FloatTensor([[5, 5, 5]])]
torch_mesh_test = torch_mesh(verts=vertecies, faces=faces, textures=None)
torch_pc_test = torch_pc(points_tensor)
torch_error = loss.point_mesh_face_distance(torch_mesh_test, torch_pc_test)
print(torch_error)
------------------------------------------------

Versions of libs:
torch==1.7.1+cu101
pytorch3d==0.4.0
Python 3.6.8

**Comment by patricklabatut** [comment]
It turns out the triangle of the provided example is actually quite small for single-precision floating point calculation. It is below some of the internal limits used in PyTorch3D's implementation of the various distance computations.

After investigation, I believe the first error in the computation for this example is introduced [there](https://github.com/facebookresearch/pytorch3d/blob/master/pytorch3d/csrc/utils/geometry_utils.cuh#L495) where the calculation is simply losing significant digits. This initial error then leads to a failed test and later to a wrong returned result.

I will thus mark this as a bug. It is however not immediately clear whether that can easily and safely be addressed.

**Comment by patricklabatut** [comment]
Meanwhile, if accurate computations are needed, the input data could simply be rescaled as a workaround. In this specific example, I believe applying a scaling factor of 1.5x to the triangle vertices and point resulted in results more in line with expectations.

**Comment by patricklabatut** [comment]
This is the same issue as #613. Please follow new developments there, I am closing this one as duplicate.

**Comment by gkioxari** [comment]
Fixed by https://github.com/facebookresearch/pytorch3d/commit/88f5d790886b26efb9f370fb9e1ea2fa17079d19
